x5.md

```diff
@@ -8,7 +8,118 @@ created: 2019-04-18
 updated:
 ---
 
-# Introduction
+# Trust Establishment Protocol Specification Draft
+## Inspiration
+- https://datprotocol.github.io/how-dat-works/
+- 
+
+:::info
+**Trust establishment deals with users verifying they are communicating with who they think they are.**
+:::
+
+## Terms Glossary
+| term | description |
+| ---- | ----------- |
+| privkey | ECDSA secp256k1 private key |
+| pubkey | ECDSA secp256k1 public key |
+| whisper key | pubkey for chat with HD derivation path m/44'/60'/0'/0/0 |
+
+
+## Information Gathering Notes
+:::danger
+how do we identify nodes (mailservers/bootnodes/etc)
+:::
+possible connections
+- client - client (not really ever, this is facilitated through all other connections)
+    - personal chat
+        - ratcheted with X3DH
+    - private group chat
+    - public chat
+- client - mailserver (statusd + ???)
+    - a mailserver identifies itself by ???
+- client - whisper node (statusd)
+    - a node identifies itself by ???
+- client - bootnode (geth)
+    - a bootnode identifies itself by ???
+- client - ENS registry (ethereum blockchain -> default to infura)
+- client - Ethereum RPC (custom geth RPC API -> default to infura API)
+- client - IPFS (Status hosted IPFS gateway -> defaults to ???)
+
+A user in the system is a public-private key pair using the Elliptic-Curve Cryptography secp256k1 that Ethereum uses.
+- A 3-word random name is derived from the public key using the following package
+    - 
+    - This provides an associated human-readble fingerprint to the user's public key
+- A user can optionally add additional layers on top of this keypair
+    - Chosen username
+    - ENS username
+
+All messages sent are encrypted with the public key of the destination and signed by the private key of the given user using the following scheme:
+- private chat
+    - the message is encrypted with the public key of the intended user
+- private group chat
+    - the message is pairwise encrypted with each of the public keys of the members of the private group, e.g. for a group of 6 members, 5 messages will be sent. 
+- public group chat
+    - the message is encrypted with the topic 
+
+## Initial Trust Establishment Specification
+### 1. Contact Discovery
+#### 1.1 Public channels
+- Public group channels in Status are a broadcast/subscription system.  All public messages are encrypted with a symmetric key drived from the channel name, $K_{\text{pub,sym}}$, which is publicly known.
```

Comment:
TeX is nice, but is it something we want in a markdown/plaintext spec? It often doesn't render nicely and it's not clear we need it. I.e. RFCs don't usually have it. I'd rather KISS here and just use ``

Reply:
ok, will change all TeX via an issue after merge